components/expenses/PayoutBankInformationForm.js

```diff
@@ -347,33 +368,76 @@ const availableCurrenciesQuery = gqlV2/* GraphQL */ `
 
 /**
  * Form for payout bank information. Must be used with Formik.
+ *
+ * The main goal is to use this component in the Expense Flow (1) but it's also reused in:
+ *
+ * - Collective onboarding, AcceptContributionsOurselvesOrOrg.js (2)
+ * - In Collective/Host settings -> Receiving Money, BankTransfer.js (3)
+ *
+ * In (1) we pass the host where the expense is submitted and fixedCurrency is never set.
+ *   * If Wise is configured on that host, `availableCurrencies` should normally be available.
+ *   * If it's not, we'll have to fetch `availableCurrencies` from the Platform Wise account
+ *
+ * In (2) and (3), we never pass an `host` and `fixedCurrency` is sometimes set.
+ *   * If `fixedCurrency` is set, we don't need `availableCurrencies`
+ *   * If `fixedCurrency` is not set, we'll fetch `availableCurrencies` from the Platform Wise account
  */
 const PayoutBankInformationForm = ({ isNew, getFieldName, host, fixedCurrency, ignoreBlockedCurrencies, optional }) => {
   const { data, loading } = useQuery(availableCurrenciesQuery, {
     context: API_V2_CONTEXT,
-    variables: { slug: host?.transferwise ? host.slug : TW_API_COLLECTIVE_SLUG, ignoreBlockedCurrencies },
-    // Skip fetching/loading if the currency is fixed or availableCurrencies was pre-loaded
-    skip: Boolean(fixedCurrency || host.transferwise?.availableCurrencies),
+    variables: { slug: TW_API_COLLECTIVE_SLUG, ignoreBlockedCurrencies },
```

Comment:
🔴 We should still prioritize `host.slug` if it's connected to Wise. The available currencies might change depending on the host.

Reply:
@kewitz : I've added the `host.slug` if `host.transferwise` is defined; as we had before. So if I am getting this correctly this only applies to the case where the host has Wise configured but `host.transferwise?.availableCurrencies` is not defined (as well as `fixedCurrency`). Added a comment as well to explain the case. Let me know if that looks good. 🤔

Reply:
I disagree with the change, it was clearly written in the comment that we expect `availableCurrencies` to be already available at this point. If it's not, there is a problem with it and re-fetching will not help.

Reply:
We should check for the host existence in the actual exported component `PayoutBankInformationForm` and not in  `DetailsForm`.
At that level, we'll always be using `TW_API_COLLECTIVE_SLUG` Wise account.

Reply:
> We should check for the host existence in the actual exported component PayoutBankInformationForm and not in DetailsForm. At that level, we'll always be using TW_API_COLLECTIVE_SLUG Wise account.

@kewitz I don't understand, is it a justification for the merged behavior or a proposal for something else?

Reply:
@znarf it is the justification why we need always prioritize the host we pass through the property.
I think the expected behavior is:
- If a host is passed down through the property, use it.
  - A host can be passed with `availableCurrencies` pre-loaded, to skip the first loading state of the component.
  - If host from props is not connected to Wise, use platform slug.
- If no host is passed down through the property, use platform slug.
  - If platform slug from props is not connected to Wise, display warning message.

Reply:
> A host can be passed with availableCurrencies pre-loaded

It's not how it works practically, in case (1) `availableCurrencies` is **ALWAYS** fetched before and passed. If it's not, then there is a problem with the Host Wise account and re-fetching will **NOT** help.

Reply:
I was under the impression that even if Wise is configured on the host `availableCurrencies` might be not available and thus thought it would be prudent to re-fetch. But now when I look at it; that doesn't seem necessary as if the first fetch is not successful there's little chance another fetch could do the trick. So I've went ahead and reverted the last change in the following PR. 

https://github.com/opencollective/opencollective-frontend/pull/6915

> red_circle We should still prioritize `host.slug` if it's connected to Wise. The available currencies might change depending on the host.

@kewitz : For your initial comment above I think we should be good here actually. As if the host has Wise configured correctly the the `availableCurrencies` are there so we skip the loading of `data`; 

https://github.com/opencollective/opencollective-frontend/blob/bd138f1f646d58860904b5b531fe998d0624cf14/components/expenses/PayoutBankInformationForm.js#L387-L393

and thus the `wiseHost` will actually be `host`;

https://github.com/opencollective/opencollective-frontend/blob/bd138f1f646d58860904b5b531fe998d0624cf14/components/expenses/PayoutBankInformationForm.js#L402-L404